src/Client.ts

```diff
@@ -5,22 +5,32 @@ import {
 } from '@kontent-ai/delivery-sdk';
 import packageInfo from '../package.json';
 import { selectedProjectCookieName } from './const';
-import { defaultProjectId } from './Utilities/SelectedProject';
+import validator from 'validator';
 
 const sourceTrackingHeaderName = 'X-KC-SOURCE';
 
-// environment variables
-const projectId = process.env.REACT_APP_PROJECT_ID || '';
 const previewApiKey = process.env.REACT_APP_PREVIEW_API_KEY || '';
 
 const cookies = new Cookies(document.cookie);
 
-let currentProjectId = projectId || cookies.get(selectedProjectCookieName);
-if (currentProjectId) {
-  cookies.set(selectedProjectCookieName, currentProjectId, { path: '/' });
-} else {
-  currentProjectId = defaultProjectId;
-}
+const getProjectId = (): string => {
+  const projectId = process.env.REACT_APP_PROJECT_ID || '';
+
+  let currentProjectId = projectId;
+
+  if (projectId) {
+    if (!validator.isUUID(projectId)) {
+      console.error(

```

Comment:
It's good that you log an error when you find an invalid `projectId`, but then you will still pass it on to the rest of the application. In my opinion, you should return an empty string in such a case to disregard the invalid value.